client/components/suggestions/index.jsx

```diff
@@ -0,0 +1,248 @@
+			suggestions: {},
+			filterTerm: ''
+		};
+	},
+
+	componentWillMount: function() {
+		const suggestions = this.narrowDown( this.props.input );
+		this.setState( {
+			suggestions: suggestions,
+			taxonomySuggestionsArray: this.createTaxonomySuggestionsArray( suggestions ),
+			currentSuggestion: '',
+		} );
+	},
+
+	componentWillReceiveProps: function( nextProps ) {
+		if ( nextProps.input !== this.props.input ) {
+			const suggestions = this.narrowDown( nextProps.input );
+			this.setState( {
+				suggestions: suggestions,
+				taxonomySuggestionsArray: this.createTaxonomySuggestionsArray( suggestions ),
+				suggestionPosition: -1,
+				currentSuggestion: '',
+			} );
+		}
+	},
+
+	countSuggestions: function() {
+		return this.state.taxonomySuggestionsArray.length;
+	},
+
+	getSuggestionForPosition: function( position ) {
+		return this.state.taxonomySuggestionsArray[ position ];
+	},
+
+	getPositionForSuggestion: function( suggestion ) {
+		return this.state.taxonomySuggestionsArray.indexOf( suggestion );
+	},
+
+	incPosition: function() {
+		const position = ( this.state.suggestionPosition + 1 ) % this.countSuggestions();
+		this.setState( {
+			suggestionPosition: position,
+			currentSuggestion: this.getSuggestionForPosition( position ),
+		} );
+	},
+
+	decPosition: function() {
+		const position = ( this.state.suggestionPosition - 1 ) % this.countSuggestions();
+		this.setState( {
+			suggestionPosition: position,
+			currentSuggestion: this.getSuggestionForPosition( position )
+		} );
+	},
+
+	handleKeyEvent: function( event ) {
+		switch ( event.key ) {
+			case 'ArrowDown' :
+				this.incPosition();
+				event.preventDefault();
+				break;
+			case 'ArrowUp' :
+				this.decPosition();
+				event.preventDefault();
+				break;
+			case 'Enter' :
+				if ( this.state.suggestionPosition !== -1 ) {
+					this.props.suggest( this.state.currentSuggestion );
+				}
+				break;
+		}
+	},
+
+	onMouseDown: function( event ) {
+		this.props.suggest( event.target.textContent );
+	},
+
+	onMouseOver: function( event ) {
+		this.setState( {
+			suggestionPosition: this.getPositionForSuggestion( event.target.textContent ),
+			currentSuggestion: event.target.textContent,
+		} );
+	},
+
+	removeEmptySuggestions: function( suggestions ) {
+		const hasValues = x => x.length > 0;
+		return pickBy( suggestions, hasValues );
+	},
+
+	narrowDown: function( input ) {
+		const [ taxonomy, filter ] = input.split( ':' );
+		if ( taxonomy === '' ) {
+			// empty string or just ":" or ":filter" -
+			// TODO: just show welcome screen
+			return {};
+		}
+
+		let limit = 3; //default limit, changed to 5 for single taxonomy
+		let terms; //terms that we will use to create suggestions
+		let filterTerm; // part of input that will be used for filtering
+
+		if ( filter !== undefined ) {
+			// this means that we have at least taxonomy:
+			// so check if this is a correct taxonomy
+			if ( this.props.terms.hasOwnProperty( taxonomy ) ) {
```

Comment:
`includes` searches only for values so it will not spot taxonomy.
we have to use 'has' here.